### PR TITLE
Update namespace finalizer to delete RoleBindingRestrictions

### PR DESCRIPTION
**Pull request by enj**
https://bugzilla.redhat.com/show_bug.cgi?id=1435132

[test]

Signed-off-by: Monis Khan <mkhan@redhat.com>

Who wants to review 😄 

cc @ncdc @mfojtik @derekwaynecarr @deads2k @liggitt @smarterclayton

### CODE REVIEW

File: pkg/project/controller/controller_test.go [hunk 1]
```diff
@@ -38,35 +38,63 @@ func TestSyncNamespaceThatIsTerminating(t *testing.T) {
...
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "roles"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "routes"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "templates"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "builds"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "namespace"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "deploymentconfig"}, "", kapi.ListOptions{}),
-		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "v1", Resource: "egressnetworkpolicy"}, "", kapi.ListOptions{}),
+		namespaceFinalize,
+		core.NewListAction(unversioned.GroupVersionResource{Group: "", Version: "", Resource: "buildconfigs"}, testNamespace.Name, kapi.ListOptions{}),
```
Comment (mfojtik) [inline pkg/project/controller/controller_test.go]: why removing version?

**Comment by mfojtik** [comment]
lgtm

File: pkg/project/controller/controller_test.go (hunk 1 above)
Comment (enj) [inline pkg/project/controller/controller_test.go]: The fake clients only keep track of resource, they always pass the empty string for group and version.

Comment (deads2k) [inline pkg/project/controller/controller_test.go]: This isn't obvious.  Is there a smaller change for a backport?  Especially on a test for a controller we're going to remove post-rebase.

Comment (enj) [inline pkg/project/controller/controller_test.go]: The old test did not check anything but the total number of actions (i.e. it was useless) so we have not lost anything.

Comment (deads2k) [inline pkg/project/controller/controller_test.go]: > The old test did not check anything but the total number of actions (i.e. it was useless) so we have not lost anything.

If it was useless, then you needed no changes to pass, right?  I'd sort of expect to see *some* change here even if you didn't have to do all this to have equivalence.

Comment (enj) [inline pkg/project/controller/controller_test.go]: Yes, at the bare minimum you had to copy paste one of the NewListAction lines.

Comment (deads2k) [inline pkg/project/controller/controller_test.go]: > Yes, at the bare minimum you had to copy paste one of the NewListAction lines.

For code that we know we're going to delete, that has a fixed point where we're going to delete it, that we know we're going to backport, I'm not sure I'd make the larger change.

@mfojtik have you reasoned through the test to see if it still works?  I haven't, but if we really want to go to the effort and we know this change is ok then actually checking is clearly better.

**Comment by enj** [comment]
@deads2k @liggitt I can has merge now?

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**